lib/SILGen/SILGenExpr.cpp

```diff
@@ -1590,8 +1587,6 @@ RValue RValueEmitter::visitOptionalTryExpr(OptionalTryExpr *E, SGFContext C) {
     return RValue(SGF, E, SGF.emitManagedRValueWithCleanup(arg, optTL));
   }
 
-  optInit->finishInitialization(SGF);
-
```

Comment:
Now that we're using `emitExprInto` above, it already calls `finishInitialization`. Calling it again here causes an assert.

(Note that `emitExprInto` always happens on the `isByAddress == true` branch, and this `finishInitialization` call also only happens when `isByAddress == true` due to the `return` in the `if` block above.)